src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -7,34 +7,35 @@ using LinearAlgebra.BLAS: gemv, gemv!, gemm!, trsm!, axpy!, ger!
 
 function rrule(::typeof(svd), X::AbstractMatrix{<:Real})
     F = svd(X)
-    function svd_pullback(Ȳ::NamedTuple{(:U,:S,:V)})
+    function svd_pullback(Ȳ::Composite{<:SVD})
         ∂X = @thunk(svd_rev(F, Ȳ.U, Ȳ.S, Ȳ.V))
         return (NO_FIELDS, ∂X)
     end
     return F, svd_pullback
 end
 
-function rrule(::typeof(getproperty), F::SVD, x::Symbol)
+function rrule(::typeof(getproperty), F::T, x::Symbol) where T <: SVD
     function getproperty_svd_pullback(Ȳ)
-        if x === :U
-            ∂ = @thunk((; U=Ȳ, S=(zero(F.S)), V=(zero(F.V))))
+        C = Composite{T}
+        ∂F = if x === :U
+            C(U=Ȳ,)
         elseif x === :S
-            ∂ = @thunk((; U=(zero(F.U)), S=Ȳ, V=(zero(F.V))))
+            C(S=Ȳ,)
         elseif x === :V
-            ∂ = @thunk((; U=(zero(F.U)), S=(zero(F.S)), V=Ȳ))
+            C(V=Ȳ,)
         elseif x === :Vt
             # TODO: This could be made to work, but it'd be a pain
+            # https://github.com/JuliaDiff/ChainRules.jl/issues/106
             throw(ArgumentError("Vt is unsupported; use V and transpose the result"))
         end
-
-        update = (X̄::NamedTuple{(:U,:S,:V)}) -> _update!(X̄, ∂, x)
-        ∂F = InplaceableThunk(∂, update)
         return NO_FIELDS, ∂F, DoesNotExist()
     end
     return getproperty(F, x), getproperty_svd_pullback
 end
 
-function svd_rev(USV::SVD, Ū::AbstractMatrix, s̄::AbstractVector, V̄::AbstractMatrix)
+# TODO: decide if we want to type constrain these inputs to `Union{Zero, T}`
```

Comment:
This would probably be a good move. We know we won't get thunks because `getproperty` should do `unthunk`ing for us, so this seems appropriate. Might be better off with `AbstractZero` though.

Reply:
I am happy to duck-type it i think

Reply:
I don't feel strongly enough either way to make a case for either option.